src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -846,24 +839,29 @@ private void ReportDiagnostic(ErrorCode errorCode, SyntaxNode syntaxNode, params
             }
         }
 
-        private void InheritNullableStateOfTrackableStruct(TypeSymbol targetType, int targetSlot, int valueSlot, bool isDefaultValue, int slotWatermark)
+        private void InheritNullableStateOfTrackableStruct(TypeSymbol targetType, int targetSlot, int valueSlot, bool isDefaultValue, int stopAtSlot = -1)
         {
             Debug.Assert(targetSlot > 0);
             Debug.Assert(EmptyStructTypeCache.IsTrackableStructType(targetType));
 
+            if (stopAtSlot < 0)
+            {
+                stopAtSlot = targetSlot;
```

Comment:
>stopAtSlot = targetSlot; [](start = 16, length = 24)

Do we still have an issue that we might be overwriting the state before an attempt to copy it?

Reply:
`TrackNullableStateForAssignment` returns early if `targetSlot == valueSlot`.

---
In reply to: [258287547](https://github.com/dotnet/roslyn/pull/33455#discussion_r258287547) [](ancestors = 258287547)